Providers/Profile.swift

```diff
@@ -607,15 +627,17 @@ public class BrowserProfile: Profile {
         }
 
         private func endSyncing() {
+            // loop through status's and fill sync state
             syncLock.lock()
             defer { syncLock.unlock() }
             log.info("Ending all queued syncs.")
+            syncDisplayState = displayStateForEngineResults(syncReducer?.terminal.value)
```

Comment:
Also, in either case, >>>/>>== invoke `endSyncing` in the failure path since both of those operators by pass the callback if failed.